### PR TITLE
HBASE-22238 Fix TestRpcControllerFactory

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 166 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ HBASE-21512 Compile Tests _ |
| +1 | mvninstall | 254 | HBASE-21512 passed |
| +1 | compile | 25 | HBASE-21512 passed |
| +1 | checkstyle | 13 | HBASE-21512 passed |
| +1 | shadedjars | 256 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 38 | HBASE-21512 passed |
| +1 | javadoc | 14 | HBASE-21512 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 243 | the patch passed |
| +1 | compile | 25 | the patch passed |
| +1 | javac | 25 | the patch passed |
| +1 | checkstyle | 13 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 261 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 505 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 45 | the patch passed |
| +1 | javadoc | 14 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 446 | hbase-endpoint in the patch failed. |
| +1 | asflicense | 11 | The patch does not generate ASF License warnings. |
| | | 2405 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.coprocessor.TestRegionServerCoprocessorEndpoint |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/147 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux d173a24ec8ed 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | HBASE-21512 / f47f6dfe51 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/2/artifact/out/patch-unit-hbase-endpoint.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/2/testReport/ |
| Max. process+thread count | 3044 (vs. ulimit of 10000) |
| modules | C: hbase-endpoint U: hbase-endpoint |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 287 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ HBASE-21512 Compile Tests _ |
| +1 | mvninstall | 285 | HBASE-21512 passed |
| +1 | compile | 23 | HBASE-21512 passed |
| +1 | checkstyle | 11 | HBASE-21512 passed |
| +1 | shadedjars | 263 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 42 | HBASE-21512 passed |
| +1 | javadoc | 14 | HBASE-21512 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 268 | the patch passed |
| +1 | compile | 24 | the patch passed |
| +1 | javac | 24 | the patch passed |
| +1 | checkstyle | 10 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 252 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 501 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 51 | the patch passed |
| +1 | javadoc | 12 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 565 | hbase-endpoint in the patch failed. |
| +1 | asflicense | 9 | The patch does not generate ASF License warnings. |
| | | 2686 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.coprocessor.TestRegionServerCoprocessorEndpoint |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/147 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux e32c37092806 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | HBASE-21512 / f47f6dfe51 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/1/artifact/out/patch-unit-hbase-endpoint.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/1/testReport/ |
| Max. process+thread count | 2923 (vs. ulimit of 10000) |
| modules | C: hbase-endpoint U: hbase-endpoint |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-endpoint/src/test/java/org/apache/hadoop/hbase/client/TestRpcControllerFactory.java [hunk 1]
```diff
@@ -85,32 +86,15 @@ public HBaseRpcController newController(final List<CellScannable> cellIterables)
 
     private static Multiset<Integer> GROUPED_PRIORITY = ConcurrentHashMultiset.create();
     private static AtomicInteger INT_PRIORITY = new AtomicInteger();
-    private static AtomicInteger TABLE_PRIORITY = new AtomicInteger();
```
Comment (openinx) [inline hbase-endpoint/src/test/java/org/apache/hadoop/hbase/client/TestRpcControllerFactory.java]: For what purpose, we need to remove the table priroirty counter ?   don't understand here ...

Comment (Apache9) [reply hbase-endpoint/src/test/java/org/apache/hadoop/hbase/client/TestRpcControllerFactory.java]: We do not call the setPriority(TableName) method any more in async client, we will decide the priority outside the RpcController and then set with the setPriority(int) method. This is why we need to change the UT itself.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 47 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ HBASE-21512 Compile Tests _ |
| +1 | mvninstall | 264 | HBASE-21512 passed |
| +1 | compile | 25 | HBASE-21512 passed |
| +1 | checkstyle | 14 | HBASE-21512 passed |
| +1 | shadedjars | 291 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 38 | HBASE-21512 passed |
| +1 | javadoc | 14 | HBASE-21512 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 248 | the patch passed |
| +1 | compile | 27 | the patch passed |
| +1 | javac | 27 | the patch passed |
| +1 | checkstyle | 14 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 272 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 523 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 44 | the patch passed |
| +1 | javadoc | 14 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 252 | hbase-endpoint in the patch passed. |
| +1 | asflicense | 11 | The patch does not generate ASF License warnings. |
| | | 2168 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/147 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux f5a601f0921e 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | HBASE-21512 / d8fe1bb77c |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/3/testReport/ |
| Max. process+thread count | 3000 (vs. ulimit of 10000) |
| modules | C: hbase-endpoint U: hbase-endpoint |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-147/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**